src/zarr/core/metadata/v2.py

```diff
@@ -191,18 +213,11 @@ def from_dict(cls, data: dict[str, Any]) -> ArrayV2Metadata:
 
         _data = {k: v for k, v in _data.items() if k in expected}
 
-        return cls(**_data)
+        return cls(**cast(Mapping[str, Any], _data))
```

Comment:
Why does this need the new cast?

Reply:
Not entirely sure what the type inference was choking on there - been a bit since I was tangling with `mypy` here but I'll double check

Reply:
Was able to remove this. It was the result of me trying to be consistent with the superclass `Metadata` in the type signature of `from_dict` and the fact that dicts are invariant (while we need covariance for JSON being recognized as Any). I've reverted that change (though at some point it might be good to button that down)